### ISSUE TITLE
NavigationLink crashes on pop

### ISSUE THREAD

**Issue by nferruzzi** [apptekstudios/ASCollectionView#54]
I got a crash adding a simple `NavigationLink` in the PhotoGridScreen

```
	var section: ASCollectionViewSection<SectionID>
	{
		ASCollectionViewSection(
			id: 0,
			data: data,
			onCellEvent: onCellEvent,
			onDragDropEvent: onDragDropEvent,
			itemProvider: { item in
				//Example of returning a custom item provider (eg. to support drag-drop to other apps)
				NSItemProvider(object: item.url as NSURL)
		})
		{ item, state in
            NavigationLink(destination: Text("Hello")) {
                ZStack(alignment: .bottomTrailing)
                {
                    GeometryReader
                    { geom in
                        ASRemoteImageView(item.squareThumbURL)
                            .aspectRatio(1, contentMode: .fill)
                            .frame(width: geom.size.width, height: geom.size.height)
                            .clipped()
                            .opacity(state.isSelected ? 0.7 : 1.0)
                    }

                    if state.isSelected
                    {
                        ZStack
                        {
                            Circle()
                                .fill(Color.blue)
                            Circle()
                                .strokeBorder(Color.white, lineWidth: 2)
                            Image(systemName: "checkmark")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                        }
                        .frame(width: 20, height: 20)
                        .padding(10)
                    }
                }
            }
		}
	}
```

Tapping works but when I tap the "<" back button it crashes with this error

```
2019-11-19 18:42:40.641781+0100 Inks[24522:5610591] *** Assertion failure in -[UINavigationController popToViewController:transition:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKitCore_Sim/UIKit-3900.12.16/UINavigationController.m:8129
2019-11-19 18:42:40.694210+0100 Inks[24522:5610591] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Tried to pop to a view controller that doesn't exist.'```

How should I set up my project to avoid such a thing?  or how would you think I should present a child screen from the collection view ?

thank you in advance

**Comment by apptekstudios** [comment]
Sounds like another SwiftUI bug :(
Here’s the same bug happening when a navigation link is used in the nav bar: https://stackoverflow.com/questions/58404725/why-does-my-swiftui-app-crash-when-navigating-backwards-after-placing-a-navigat

I’ll have a look later today if I can find a satisfactory workaround 👍

**Comment by apptekstudios** [comment]
Just tested on iOS 13.1, and the above code [works perfectly.](https://drive.google.com/file/d/1KTU82OQa_yXB5jIpHi1Km6hNWy_JRyiR/view?usp=sharing) This is a bug introduced with 13.2 and relates to the link I posted above.

Unfortunately, on testing the 13.3 beta it's still present - I'm going to try and come up with an alternative solution - will keep you posted 👍

**Comment by nferruzzi** [comment]
I start believing apple fails hard in automated tests, we should be paid for being their beta testers! thanks for looking for an alternative :)

**Comment by apptekstudios** [comment]
I've created a little library designed to work around this bug 🤓
[ASNavigationView](https://github.com/apptekstudios/ASNavigationView)
It is very minimal currently, but might allow for some useful navigation features in the future!  

Demo usage in the PhotoGrid screen on this branch: https://github.com/apptekstudios/ASCollectionView/tree/navigationFix

Hopefully this is helpful! It's frustrating that these bugs are making it into the public releases of iOS...

Ps. thankyou for your generous coffee donation, it definitely helps motivate me figuring out these bugs 😄

**Comment by nferruzzi** [comment]
I'm happy you got the coffee and the bug as a motivation to make a new library ;) !! Ill try it and see if it avoid the crash on my project too :) thank you and let us know if you need more caffeine :)

**Comment by apptekstudios** [comment]
@nferruzzi Just pushed a newer version of ASNavigationView. It allows for some more advanced navigation, such as popping back to a specific screen :) 

I'm away for the weekend with no laptop so won't be able to do any further work until monday, hopefully this is useful to you 😄

**Comment by tarasis** [comment]
Doesn't appear to build. Just checked it out to test, get 4 compile errors:

`ASNavigationButton(screenName: "PHOTOGRID", destination: PhotoGridScreen())
				{
					Image(systemName: "1.square.fill")
					Text("Photo grid (with edit mode, selection)")
				}`
Generates: 

> Cannot invoke initializer for type 'ASNavigationButton<_, _>' with an argument list of type '(screenName: String, destination: PhotoGridScreen, @escaping () -> ())'

The two calls to  `ASNavigationPopToRootButton` generate  
> Use of unresolved identifier 'ASNavigationPopToRootButton'

If I search the project, I see ASNavigationButton in the Framework>Demo>ASCollectionViewDemo, but its not in the ProjectNavigator tree. (likewise ASNavigationState).

VStack line in 35 in PhotoGridScreen.swift generates

> Type 'CGFloat?' does not conform to protocol 'ExpressibleByIntegerLiteral'

If I add ASNavigationButton to the Project, and try and build, the other errors disappear but I get 3 new errors in ASNavigationButton, all the same

`@Environment(\.dynamicNavState) var dynamicNavState` 
which Xcode flags as 

> 'dynamicNavState' is inaccessible due to 'internal' protection level

**Comment by apptekstudios** [comment]
@tarasis I’ll check what’s happening when I get back from holiday tomorrow, but it sounds as though the framework isn’t being included in the demo project target - I may have forgotten to commit the xcodeproj file!

**Comment by apptekstudios** [comment]
@tarasis I had forgotten to push my last local commit to github sorry! Should be working now 👍

**Comment by tarasis** [comment]
@apptekstudios can confirm it now works, cheers👌

**Comment by apptekstudios** [comment]
This was **resolved in 13.3** so closing the issue 😄

There's currently a **simulator-only** bug that means ANY SwiftUI navigation link (unrelated to ASCollectionView) seems to only work once, however the behaviour is **normal on a real device.**

Nonetheless I've still got a more advanced version of ASNavigationView in the works that should solve issues with deep-linking etc in SwiftUI. It's working brilliantly in our app but still needs a lot of documentation and refactoring.
